pomchecker-cli/src/main/java/org/kordamp/maven/checker/cli/internal/PomParser.java

```diff
@@ -17,136 +17,59 @@
-        ClassWorld classWorld =
-                new ClassWorld("plexus.core", Thread.currentThread().getContextClassLoader());
-        ContainerConfiguration containerConfiguration =
-                new DefaultContainerConfiguration()
-                        .setClassWorld(classWorld)
-                        .setRealm(classWorld.getClassRealm("plexus.core"))
-                        .setClassPathScanning(PlexusConstants.SCANNING_INDEX)
-                        .setAutoWiring(true)
-                        .setJSR250Lifecycle(true)
-                        .setName("pom-reader");
-        try {
-            PlexusContainer container = new DefaultPlexusContainer(containerConfiguration);
-            try (Context context = Runtimes.INSTANCE.getRuntime().create(ContextOverrides.create().withUserSettings(true).build())) {
-                return createMavenProject(pomFile, context, container);
-            }
-        } catch (PlexusContainerException ex) {
-            throw new IllegalStateException(ex);
-        }
-    }
-
-    private static MavenProject createMavenProject(File pomFile, Context context, PlexusContainer plexusContainer) {
-        try {
-            MavenExecutionRequest mavenExecutionRequest = new DefaultMavenExecutionRequest();
-            ProjectBuildingRequest projectBuildingRequest =
-                mavenExecutionRequest.getProjectBuildingRequest();
-
-            projectBuildingRequest.setRepositorySession(context.repositorySystemSession());
-            projectBuildingRequest.setRemoteRepositories(context.remoteRepositories()
-                    .stream().map(PomParser::toArtifactRepository).collect(Collectors.toList()));
-            // Profile activation needs properties such as JDK version
-            Properties properties = new Properties(); // allowing duplicate entries
-            properties.putAll(projectBuildingRequest.getSystemProperties());
-            properties.putAll(detectOsProperties());
-            properties.putAll(System.getProperties());
-            projectBuildingRequest.setSystemProperties(properties);
-
-            ProjectBuilder projectBuilder = plexusContainer.lookup(ProjectBuilder.class);
-            ProjectBuildingResult projectBuildingResult =
-                projectBuilder.build(pomFile, projectBuildingRequest);
-            return projectBuildingResult.getProject();
-        } catch (ComponentLookupException | ProjectBuildingException ex) {
-            throw new IllegalStateException(ex);
+    /**
+     * Creates {@link MavenProject} object carrying "raw" and "effective" models of file pointed at.
+     * <p>
+     * Important: to successfully build an effective model for given POM file points at, all the relevant POMs
+     * (parent, BOMs imported, etc) MUST BE RESOLVABLE. Hence, there is possibility to provide "extra" remote
+     * repositories.
+     */
+    public static MavenProject createMavenProject(File pomFile, RemoteRepository... repositories) {
```

Comment:
Does this resolved POMs where certain properties are activated via profiles? The old code made sure to setup required activation points.

Reply:
Well, MIMA collects in same way the properties as Maven. OTOH, am really unsure what "os.detected.xxx" are, certainly are not set by Maven. Are you targeting here some specific plugin maybe?

Reply:
Right. That's why those `oss.detected.xxx` properties were in place, to make sure platform specific profiles were activated during resolution if needed. This means this PR is not behavior compatible.